packages/components/src/Button.js

```diff
@@ -31,14 +42,16 @@ const Button = ({
       className,
     )}
     disabled={disabled}
-    onClick={onClick}
+    onClick= {category === 'loader' ? startLoader : onClick}
```

Comment:
Since these are pure components we don't have to worry about implementing their behavior just yet. I suggest something like the following:

```js
// anyParentComponent.js
// `isLoadingUsers` is a property in the parent component's state that's set to `true` when it requests for something.
// this way the parent "controls" the button component

<Button loading={this.state.isLoadingUsers}>cool button</Button>
```

And inside the button we do:
```diff
  {icon && <Icon className="button__icon" name={icon} />}
+ {loading ? <Loader /> : children}
  // when the button has the "loading" state it shows the loader, otherwise, it shows its text
</button>
```